### PR TITLE
Bank Icons add for payment methods 

### PR DESCRIPTION
**Pull request by Santhosh-Sellavel**
### Details
Added bank icons for cards & bank account payment options

### Fixed Issues

$ #4398 

### Tests & QA Steps
1. Go to Settings -> Payments
2. Your Card & Bank payment methods should be their respective bank icons. Or should show our default bank or card icons.

Note: Refer to screenshot & ensure default icons & Expensify are in small size & bank logos slightly bigger as shown.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screenshot 2021-09-03 at 10 07 41 PM

#### Mobile Web

Simulator Screen Shot - iPhone 11 - 2021-09-03 at 22 11 55

#### Desktop
Screenshot 2021-09-03 at 10 08 22 PM

#### iOS

Simulator Screen Shot - iPhone 12 - 2021-09-03 at 22 08 48

#### Android

Screenshot_1630687226

### CODE REVIEW

**Comment by Santhosh-Sellavel** [comment]
@stitesExpensify I see lint error will fix those asap, please ignore it for time being, thanks!

File: assets/images/bankicons/discover.svg
Comment (stitesExpensify) [inline assets/images/bankicons/discover.svg:1]: Woah wtf this svg is 13000 lines!?

File: src/components/Icon/BankIcons.js
Comment (stitesExpensify) [inline src/components/Icon/BankIcons.js:60]: Wouldn't `.includes('discover')` cover all 3 cases here (and in the other similar ones)?

**Review by stitesExpensify — CHANGES_REQUESTED** [review]
Looking good! What is the problem you mentioned with Fidelity and Citizens bank [here](https://github.com/Expensify/App/issues/4398#issuecomment-898685281)? Can you give me an example?

File: src/components/Icon/BankIcons.js
Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js:60]: There's a bank Discovery

File: assets/images/bankicons/discover.svg
Comment (Santhosh-Sellavel) [reply assets/images/bankicons/discover.svg:1]: Screenshot 2021-08-18 at 1 11 13 AM

That's the bigger file

**Comment by Santhosh-Sellavel** [comment]
> Looking good! What is the problem you mentioned with Fidelity and Citizens bank [here]
(https://github.com/Expensify/App/issues/4398#issuecomment-898685281)? Can you give me an example?

@stitesExpensify 
For Fidelity, what banks to take from here. Look at the namings I don't know whether they are the same or different organizations
Screenshot 2021-08-18 at 1 16 20 AM

For Citizens bank, 
Screenshot 2021-08-18 at 1 14 28 AM

Minified Mapping results for fidelity & Citizen's Bank have a look at it.
```
{"fidelity":[{"bankName":"FIDELITY BANK","assetName":"fidelity"},{"bankName":"The Fidelity Bank (NC) - Personal","assetName":"fidelity"},{"bankName":"Fidelity","assetName":"fidelity"},{"bankName":"FIDELITY CO-OP BANK","assetName":"fidelity"},{"bankName":"FIDELITY DEP DISCOUNT BK","assetName":"fidelity"},{"bankName":"THE FIDELITY BANK","assetName":"fidelity"},{"bankName":"FIDELITY BANK AND TRUST","assetName":"fidelity"},{"bankName":"FIDELITY NATIONAL BANK","assetName":"fidelity"},{"bankName":"FIRST FIDELITY BANK  N.A.","assetName":"fidelity"},{"bankName":"FIDELITY BANK OF TEXAS","assetName":"fidelity"},{"bankName":"First Fidelity Bank (OK & AZ)","assetName":"fidelity"},{"bankName":"FIDELITY FED SAVINGS & LOA","assetName":"fidelity"},{"bankName":"Fidelity Bank (LA)","assetName":"fidelity"},{"bankName":"Fidelity Bank (Georgia) - Business","assetName":"fidelity"},{"bankName":"Fidelity Bank (Georgia)","assetName":"fidelity"},{"bankName":"Fidelity Bank","assetName":"fidelity"},{"bankName":"First Fidelity Bank","assetName":"fidelity"},{"bankName":"Fidelity Charitable Gift Fund","assetName":"fidelity"},{"bankName":"Fidelity Bank (IA)","assetName":"fidelity"},{"bankName":"United Fidelity Bank - Personal","assetName":"fidelity"},{"bankName":"FIRST FIDELITY BANK","assetName":"fidelity"},{"bankName":"Fidelity Bank & Oklahoma Fidelity Bank - Personal Online Banking","assetName":"fidelity"},{"bankName":"Fidelity Bank (PA) - Personal","assetName":"fidelity"}],"citizens-bank":[{"bankName":"citizensbankbusiness","assetName":"citizens-bank"},{"bankName":"Citizens Bank","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK NA","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK  NATIONAL ASSOCIATION","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF LAFAYETTE","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF MUKWONAGO","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK & TRUST CO","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF PHILADELPHIA","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF EDMOND","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK & TRUST OF CHILLICOTHE","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK MINNESOTA","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK & TRUST CO.","assetName":"citizens-bank"},{"bankName":"Citizens Bank & Trust (MO)","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK AND TRUST CO","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF CHATSWORTH","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK AND TRUST","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF ADA","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK AND TRUST COMPANY","assetName":"citizens-bank"},{"bankName":"Citizens Bank Minnesota","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK & TRUST BIG TIMBER","assetName":"citizens-bank"},{"bankName":"Citizens Bank & Trust Company","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF FARMINGTON","assetName":"citizens-bank"},{"bankName":"Citizens Bank (Swainsboro, Laurens County)","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF KANSAS  N.A.","assetName":"citizens-bank"},{"bankName":"Citizens Bank of Lafayette","assetName":"citizens-bank"},{"bankName":"Citizens Bank & Trust","assetName":"citizens-bank"},{"bankName":"Citizens Bank of Mukwonago","assetName":"citizens-bank"},{"bankName":"Citizens Bank & Trust Company (Big Timber, Montana)","assetName":"citizens-bank"},{"bankName":"Citizens Bank & Trust Company (VA)","assetName":"citizens-bank"},{"bankName":"Citizens Bank & Trust (Alabama)","assetName":"citizens-bank"},{"bankName":"Citizens Bank of Edmond","assetName":"citizens-bank"},{"bankName":"Citizens Bank of Kansas","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF KENTUCKY, INC","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK, NATIONAL ASSOCIATION","assetName":"citizens-bank"},{"bankName":"Citizens Bank of Georgia","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF KANSAS","assetName":"citizens-bank"},{"bankName":"Citizens Bank & Trust (Florida)","assetName":"citizens-bank"},{"bankName":"Citizens Bank & Trust Co. (Vivian, Louisiana)","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK OF LAS CRUCES","assetName":"citizens-bank"},{"bankName":"CITIZENS BANK & TRUST COMPANY","assetName":"citizens-bank"}]}
```

**Comment by Santhosh-Sellavel** [comment]
1. `Citizens Bank & Trust  & Citizens Bank of  *`  Is it the same bank or different?
2.  `Fidelity` is a too generic name, there are different bank name variations. I don't whether they are the same bank or different. 
See the names [here](https://github.com/Expensify/App/pull/4653#issuecomment-900587446) give me some input what to do with this @stitesExpensify

**Comment by stitesExpensify** [comment]
> Citizens Bank & Trust & Citizens Bank of * Is it the same bank or different?

Same bank

> Fidelity is a too generic name, there are different bank name variations. I don't whether they are the same bank or different.

I think it's all the same.

Let's just assume that anything with "citizens bank" in it is citizens, and anything with "Fidelity" in it is fidelity. I don't think the false positives really matter if the bank is so obscure that it has one of those things in it

File: assets/images/bankicons/discover.svg
Comment (stitesExpensify) [reply assets/images/bankicons/discover.svg:1]: Asked in the thread where we generated these why that one is gigantic

**Comment by Santhosh-Sellavel** [comment]
@stitesExpensify  I don't have a real account with bank account linked.
So I need your help me with test with real time data.

I used some dummy data for to have a peek. Here is my some screenshots, is it okay do you think we need some additional styling? let me know.

#### Android
Screenshot_1629569993
#### web
Screenshot 2021-08-21 at 11 49 47 PM

Thanks!

File: src/components/Icon/BankIcons.js
Comment (stitesExpensify) [inline src/components/Icon/BankIcons.js:24]: Need a return type

File: src/components/Icon/BankIcons.js
Comment (stitesExpensify) [inline src/components/Icon/BankIcons.js:111]: So can we not get the default credit card icon anymore? We still want that for cards that don't match anything

**Comment by stitesExpensify** [comment]
https://github.com/Expensify/App/pull/4653#issuecomment-903157000

I don't think we need more styling, it looks good to me. What about you @shawnborton

**Comment by shawnborton** [comment]
Yup, I think it's good to go.

File: src/components/Icon/BankIcons.js
Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js:24]: @stitesExpensify What's the return type here can you help me here.

File: src/components/Icon/BankIcons.js
Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js:111]: We can do it since we are trying to get Icon for Bank Name by default it should be a Bank Logo. 
I'll make the changes if needed, let me know about the default card icon size whether new size or the old size. Also confirm for default bank size also.

File: src/components/Icon/BankIcons.js
Comment (stitesExpensify) [reply src/components/Icon/BankIcons.js:24]: I believe it should just be `Object`

File: src/components/Icon/BankIcons.js
Comment (stitesExpensify) [reply src/components/Icon/BankIcons.js:111]: Default bank size looks good to me, I think card can be the same.

We definitely want a card icon when there are no matches.

I think adding a param to `getBankIcon` for `isCard` and returning the default card icon would be a good solution for this

**Comment by stitesExpensify** [comment]
One more request please, can you add one more condition to `bankIcons.js` for the expensify card specifically?

```
if(bankName === 'Expensify Card') {
    return expensifyCard
}
```

expensifyCard is in `assetes/images/expensifycard.svg`


Sorry that was not in the original request. Thanks!

File: src/components/Icon/BankIcons.js
Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js:111]: Updated

**Comment by Santhosh-Sellavel** [comment]
@stitesExpensify 

I think we are good with this, then.
Screenshot 2021-09-02 at 11 18 51 AM

**Comment by Santhosh-Sellavel** [comment]
@shawnborton 
Following the same size as the Logo

**Comment by shawnborton** [comment]
It should be the same size that it was previously, it should match the
PayPal icon. The bank logos look correct though.

**Comment by shawnborton** [comment]
That looks good to me, thanks!

**Comment by Santhosh-Sellavel** [comment]
@shawnborton 
Can you confirm on the Expensify card too?
Screenshot 2021-09-02 at 3 03 13 PM

Thanks!

**Comment by shawnborton** [comment]
That looks good for now, but I can see us wanting to add a more branded
image instead of the gray icon. We can discuss in Slack?

**Comment by Santhosh-Sellavel** [comment]
> 
> expensifyCard is in `assetes/images/expensifycard.svg`
> 

@shawnborton This was the image mentioned by @stitesExpensify 

> We can discussion in Slack? 

Sure please initiate @shawnborton

**Comment by shawnborton** [comment]
Yes yes, I am saying the image is absolutely correct but now that I see the
bank logos, perhaps we will want to update our own logo. Let’s solve that
as a separate PR & discussion though, so feel free to keep marching forward
in the meantime.

File: src/components/Icon/BankIcons.js
Comment (stitesExpensify) [inline src/components/Icon/BankIcons.js:117]: Sorry for being unclear, can you move this into the top of `getAssetIcon`

**Review by stitesExpensify — CHANGES_REQUESTED** [review]
Looks great! Thanks for all of your work on this. One small change then you can take it off WIP and fill out the screenshots/tests.

Once that's done I'll approve and merge!

File: src/components/Icon/BankIcons.js
Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js:117]: Then would check be like `bankName === 'expensify card'`.

Regarding I'll provide mock screenshots on all devices, I don't have real-time data can you verify that?

Comment (stitesExpensify) [reply src/components/Icon/BankIcons.js:117]: > Then would check be like bankName === 'expensify card'.

Sounds good!

> Regarding screenshots, I'll provide mock screenshots on all devices, I don't have real-time data can you verify that?

I only have a couple of accounts on prod, but LGTM!

Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js:117]: image

Need to resize the Default Bank, Card and Expensify  as shown above as per @shawnborton suggestion

Do you feel otherwise?

Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js:117]: Refer comments
https://github.com/Expensify/App/pull/4653#issuecomment-911220887
https://github.com/Expensify/App/pull/4653#issuecomment-911441982
https://github.com/Expensify/App/pull/4653#issuecomment-911455402
https://github.com/Expensify/App/pull/4653#issuecomment-911473261

Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js:117]: Based on your final comment I'll make changes & push if needed or just update the PR tests and add screenshots. Thanks
@stitesExpensify

Comment (stitesExpensify) [reply src/components/Icon/BankIcons.js:117]: > Need to resize the Default Bank, Card and Expensify as shown above as per @shawnborton suggestion
> Do you feel otherwise?

What shawn says goes when it comes to design :smile: Please resize them in addition to moving the `bankName === 'expensify card'`

Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js:117]: Sure, will update PR by tomorrow!

Comment (Santhosh-Sellavel) [reply src/components/Icon/BankIcons.js]: updated!

**Comment by Santhosh-Sellavel** [comment]
@stitesExpensify Ready for the review!

**Review by stitesExpensify — APPROVED** [review]
Looks great!

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-09-15.  :confetti_ball:

**Pull request merged**
